src/Symfony/Component/DependencyInjection/Compiler/ReplaceAliasByActualDefinitionPass.php

```diff
@@ -39,98 +38,91 @@ public function process(ContainerBuilder $container)
         $this->compiler = $container->getCompiler();
         $this->formatter = $this->compiler->getLoggingFormatter();
 
-        foreach ($container->getAliases() as $id => $alias) {
-            $aliasId = (string) $alias;
-
+        $seenAliasTargets = array();
+        $replacements = array();
+        foreach ($container->getAliases() as $definitionID => $target) {
+            $targetId = (string) $target;
+            // Check if we've seen this target before
+            if (isset($replacements[$targetId])) {
+                $container->setAlias($definitionID, $replacements[$targetId]);
+                continue;
+            }
+            if (isset($seenAliasTargets[$targetId])) {
+                continue;
+            }
+            // Process new target
+            $seenAliasTargets[$targetId] = true;
             try {
-                $definition = $container->getDefinition($aliasId);
+                $definition = $container->getDefinition($targetId);
             } catch (InvalidArgumentException $e) {
-                throw new InvalidArgumentException(sprintf('Unable to replace alias "%s" with actual definition "%s".', $id, $alias), null, $e);
+                throw new InvalidArgumentException(sprintf('Unable to replace alias "%s" with actual definition "%s".', $definitionID, $target), null, $e);
```

Comment:
No idea :3 That's what the old code was doing though (using `$alias` instead of `$aliasId`). I'll change it.